File: Scadex.Model/Auth/Login/LoginResponse.cs
using Scadex.Core.Utils.Auth;
using Scadex.Model.Dtos.User.Queries;

namespace Scadex.Model.Auth.Login;

public class LoginResponse
{
    public IList<string>? Roles { get; set; }
    public ICollection<string> Permissions { get; set; } = new List<string>();

    public AccessToken AccessToken { get; set; } = null!;
    public Guid DeviceId { get; set; }
    public UserBaseDto User { get; set; } = null!;
}

public class LoginTrustedResponse : LoginResponse
{
    public string RefreshToken { get; set; } = null!;
}

File: Scadex.WebUI/src/models/auth/queries/loginResponse.ts
import type { AccessToken } from '@/models/auth/queries/accessToken';
import type { UserBaseDto } from '@/models/user/queries/userBaseDto';

export interface LoginResponse {
  roles: string[] | null;
  permissions: string[];
  accessToken: AccessToken;
  deviceId: string;
  user: UserBaseDto;
}
